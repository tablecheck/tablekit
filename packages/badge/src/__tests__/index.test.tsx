import { mount } from '@tablecheck/tablekit-enzyme';

import { Badge } from '../index';

beforeEach(() => {
  global.console.error = jest.fn();
});

describe('Badge', () => {
  test('should be possible to create a component', () => {
    const wrapper = mount(<Badge value={0} />);
    expect(wrapper).not.toBeUndefined();
  });

  test('should correctly apply styling', () => {
    expect(mount(<Badge value={0} />)).toMatchSnapshot();
  });

  test('should allow passing of custom className', () => {
    expect(mount(<Badge className="test-class" value={0} />)).toMatchSnapshot();
  });

  test('should allow passing of custom children', () => {
    expect(
      mount(
        <Badge>
          <p>foo</p>
        </Badge>
      )
    ).toMatchSnapshot();
  });
});
