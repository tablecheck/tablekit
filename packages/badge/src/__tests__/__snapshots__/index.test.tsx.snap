// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should allow passing of custom children 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 9999px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  width: 24px;
  height: 24px;
  padding: 0 8px;
  border: 1px solid rgba(0, 0, 0, 0.11);
  color: #323232;
}

<Badge>
  <BadgeContainer>
    <Noop />
    <div
      className="emotion-0 emotion-1"
    >
      <p>
        foo
      </p>
    </div>
  </BadgeContainer>
</Badge>
`;

exports[`Badge should allow passing of custom className 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 9999px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  width: 24px;
  height: 24px;
  padding: 0 8px;
  border: 1px solid rgba(0, 0, 0, 0.11);
  color: #323232;
}

<Badge
  className="test-class"
  value={0}
>
  <BadgeContainer
    className="test-class"
  >
    <Noop />
    <div
      className="test-class emotion-0 emotion-1"
    >
      0
    </div>
  </BadgeContainer>
</Badge>
`;

exports[`Badge should correctly apply styling 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 9999px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  width: 24px;
  height: 24px;
  padding: 0 8px;
  border: 1px solid rgba(0, 0, 0, 0.11);
  color: #323232;
}

<Badge
  value={0}
>
  <BadgeContainer>
    <Noop />
    <div
      className="emotion-0 emotion-1"
    >
      0
    </div>
  </BadgeContainer>
</Badge>
`;
